File: server/bcephal-alarm/src/main/java/com/moriset/bcephal/alarm/domain/AlarmAudience.java
/**
 * 
 */
package com.moriset.bcephal.alarm.domain;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.moriset.bcephal.domain.Persistent;

import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.SequenceGenerator;
import jakarta.persistence.Table;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;

/**
 * @author Joseph Wambo
 *
 */
@Entity(name = "AlarmAudience")
@Table(name = "BCP_ALARM_AUDIENCE")
@Data
@EqualsAndHashCode(callSuper = false)
public class AlarmAudience extends Persistent {

	private static final long serialVersionUID = -3228934629560509283L;

	@Id
	@GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "alarm_aud_seq")
	@SequenceGenerator(name = "alarm_aud_seq", sequenceName = "alarm_aud_seq", initialValue = 1,  allocationSize = 1)
	private Long id;
	
	@ToString.Exclude
	@EqualsAndHashCode.Exclude	
	@JsonIgnore
	@ManyToOne(fetch = FetchType.LAZY)
	@JoinColumn(name = "alarm")
	private Alarm alarm;
	
	private int position;

	private boolean active;
	
	@Enumerated(EnumType.STRING)
	private AlarmAudienceType audienceType;
	
	@Enumerated(EnumType.STRING)
	private EmailType emailType;

	private Long userOrProfilId;
	
	private String name;
	
	private String email;
	
	private String phone;
	
    private boolean sendEmail;
	
    public boolean sendSms;
	
    private boolean sendChat;
		
	
	public AlarmAudience() {
		setActive(true);
		setEmailType(EmailType.TO);
	}

	
	@JsonIgnore
	public AlarmAudience copy() {
		return null;
	}

	
}

File: client/bcephal-client-model/Alarms/AlarmAudience.cs
using Bcephal.Models.Base;
using Newtonsoft.Json;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Bcephal.Models.Alarms
{
   public class AlarmAudience : Persistent
    {
        public int Position { get; set; }

        public bool Active { get; set; }

        [JsonIgnore]
        public long? AlarmId { get; set; }

        public long UserOrProfilId { get; set; }

        public string Name { get; set; }

        public string Email { get; set; }

        public string Phone { get; set; }

        public bool SendEmail { get; set; }

        public bool SendSms { get; set; }

        public bool SendChat { get; set; }

        public AlarmAudienceType AudienceType { get; set; }

        public AlarmAudience()
        {
            Position = -1;
        }

        public override int CompareTo(object obj)
        {
            if (obj == null || !(obj is AlarmAudience)) return 1;
            if (this == obj) return 0;
            if (this.Id.HasValue && this.Id.Equals(((AlarmAudience)obj).Id)) return 0;
            int c = this.Position.CompareTo(((AlarmAudience)obj).Position);
            if (c != 0) return c;
            if (this.AlarmId.HasValue && ((AlarmAudience)obj).AlarmId.HasValue) return this.AlarmId.Value.CompareTo(((AlarmAudience)obj).AlarmId);
            return 1;
        }

        public AlarmAudience Copy()
        {
            AlarmAudience copy = new AlarmAudience();
            copy.Id = this.Id;
            copy.Position = this.Position;
            copy.Active = this.Active;
            copy.AlarmId = this.AlarmId;
            copy.UserOrProfilId = this.UserOrProfilId;
            copy.Name = this.Name;
            copy.Email = this.Email;
            copy.Phone = this.Phone;
            copy.SendEmail = this.SendEmail;
            copy.SendSms = this.SendSms;
            copy.SendChat = this.SendChat;
            copy.AudienceType = this.AudienceType;
            return copy;
        }

    }
}
                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         